samza-core/src/main/java/org/apache/samza/execution/JobNode.java

```diff
@@ -255,21 +239,11 @@ void addSerdeConfigs(Map<String, String> configs) {
         }
       });
 
-    // collect all key and msg serde instances for tables
```

Comment:
Ideally all serialized objects should be written to the same OutputStream in one pass to maintain their relationships upon deserialization. Is there a way to refactor TableConfigGenerator so that this is still possible?

Reply:
Perhaps the TableConfigGenerator take a list of already generated serdes?